test/test_art_span.cpp

```diff
@@ -0,0 +1,1116 @@
+// Copyright 2019-2024 Laurynas Biveinis
+
+// IWYU pragma: no_include <array>
```

Comment:
These pragmas go below global.hpp. I did not review the rest of this file :)

Reply:
Removed